modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/DoubleGaugeAdapter.java

```diff
@@ -10,33 +10,45 @@
 
 import io.opentelemetry.api.metrics.Meter;
 
+import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * DoubleGaugeAdapter wraps an otel ObservableDoubleMeasurement
  */
-public class DoubleGaugeAdapter extends AbstractInstrument<io.opentelemetry.api.metrics.ObservableDoubleMeasurement>
+public class DoubleGaugeAdapter extends AbstractInstrument<io.opentelemetry.api.metrics.ObservableDoubleGauge>
     implements
         org.elasticsearch.telemetry.metric.DoubleGauge {
 
+    private final AtomicReference<ValueWithAttributes<Double>> valueWithAttributes;
```

Comment:
Why not just do a new record here?  That will avoid boxing as well. 
`record DoubleValueWithAttribute(double value, Map<String, Object> attributes)`.